### PR TITLE
Feat/71 responsive update for the navbar

### PR DESCRIPTION
**Pull request by jiayuu-u**
### Descriptions

- Link to UI design (if possible):

### Provide screenshots or videos that the function is working.
 

https://user-images.githubusercontent.com/68804519/223590869-7590c544-69ba-499b-8509-0085c2de4bc9.mp4

### Provide screenshots of no warning, or error in the console.
image

- [x] All unit tests passing and coverage of testing is more than 90% unless special needs.
- [x] No merge conflict with the `main` branch, make sure your code is up to date with `main`.
- [x] Check PR title is aligned with your ticket.
- [x] Check your app will run without crashes.
- [x] Check if there are hard-coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] Check you are using `camelCase && PascalCase` correctly.
- [x] Remove unused comments.

### CODE REVIEW

**Review by Trent1900 — APPROVED** [review]
LGTM

**Comment by HarryuCATGREY** [comment]
Bug Report
You cannot slide the page after clicking the menu button and close the nav bar.

File: components/base/Card.tsx [hunk 1]
```diff
@@ -1,25 +1,59 @@
...
+        {direction === "column_reverse" && !large_title && (
+          <h1 className={`${fontSyne.className} global__text-xlarge`}>{title}</h1>
+        )}
+        {direction !== "column_reverse" && (
+          <h1 className={`${fontSyne.className} global__text--large`}>{title}</h1>
+        )}
+        <p className={`${fontRobotoMono.className} global__text-meta`}>{subtitle}</p>
       </div>
-    </figure>
+    </div>
```
Comment (tatoMa) [inline components/base/Card.tsx:56]: The figure is not a perfect semantic HTML tag here but it is better than a div you have changed.

**Review by tatoMa — COMMENTED** [review]
You have so many changes out of your card if I'm not wrong. PLEASE, do not do this again.

File: components/general/Navbar/FirstPageNavbar.tsx [hunk 2]
```diff
@@ -30,15 +35,26 @@ const NavBar = () => {
 
   return (
     <section className={styles.header_navbar}>
-      <section className={styles.header_navbar_link}>
+      <nav className={openNav ? styles.header_responsive_navbar : styles.header_navbar_menu}>
```
[the remark this answers is not in the document]
Comment (jiayuu-u) [reply components/general/Navbar/FirstPageNavbar.tsx:38]: Yes. The header_responsive_navbar is for mobile and the header_navbar_menu is for desktop.

File: components/general/Navbar/FirstPageNavbar.tsx [hunk 3]
```diff
@@ -30,15 +35,26 @@ const NavBar = () => {
...
         <Button aria-label="let's_talk_button">LET&apos;S TALK</Button>
-      </section>
+      </div>
+      <div
+        className={`${styles.header_navbar_menu_icon} ${
+          openNav ? styles.header_navbar_close_icon : null
+        }`}
+        onClick={handleNavBar}
+        data-testid="nav-icon"
+      >
```
[the remark this answers is not in the document]
Comment (jiayuu-u) [reply components/general/Navbar/FirstPageNavbar.tsx:53]: It's a clickable menu button. When we click, the button will change to "X" style then the menu open.

File: layouts/Header.tsx [hunk 4]
```diff
@@ -2,8 +2,21 @@ import { fontRobotoMono } from "@/pages/_app";
 import styles from "@/styles/Header.module.css";
 import NavBar from "@/components/general/Navbar/FirstPageNavbar";
 import Link from "next/link";
+import { useState, useEffect } from "react";
 
 const Header = () => {
+  const [openNav, setOpenNav] = useState(false);
```
[the remark this answers is not in the document]
Comment (jiayuu-u) [reply layouts/Header.tsx:8]: The boolean naming convention should be prefixed with is. I will change it.

File: layouts/Header.tsx [hunk 5]
```diff
@@ -2,8 +2,21 @@ import { fontRobotoMono } from "@/pages/_app";
 import styles from "@/styles/Header.module.css";
 import NavBar from "@/components/general/Navbar/FirstPageNavbar";
```
[the remark this answers is not in the document]
Comment (jiayuu-u) [reply layouts/Header.tsx:3]: It's NavBar. I will fix it

**Review by yunshen513 — APPROVED** [review]

**Pull request merged**